File: mRemoteV1/Connection/Connection.List.vb
Namespace Connection
    Public Class List
        Inherits CollectionBase

#Region "Public Properties"

        Default Public ReadOnly Property Items(Index As Object) As Info
            Get
                If TypeOf Index Is Info Then
                    Return Index
                Else
                    Return CType(List.Item(Index), Info)
                End If
            End Get
        End Property

        Public Shadows ReadOnly Property Count As Integer
            Get
                Return List.Count
            End Get
        End Property

#End Region

#Region "Public Methods"

        Public Function Add(cInfo As Info) As Info
            Me.List.Add(cInfo)
            Return cInfo
        End Function

        Public Sub AddRange(cInfo() As Info)
            For Each cI As Info In cInfo
                List.Add(cI)
            Next
        End Sub

        Public Function FindByConstantID(id As String) As Info
            For Each conI As Info In List
                If conI.ConstantID = id Then
                    Return conI
                End If
            Next

            Return Nothing
        End Function

        'Public Function Find(ByVal cInfo As Connection.Info)
        '    For Each cI As Connection.Info In List

        '    Next
        'End Function

        Public Function Copy() As List
            Try
                Return Me.MemberwiseClone
            Catch ex As Exception
            End Try

            Return Nothing
        End Function

        Public Shadows Sub Clear()
            Me.List.Clear()
        End Sub

#End Region
    End Class
End Namespace

File: mRemoteV1/Connection/ConnectionList.cs
using System;
using System.Collections;


namespace mRemoteNG.Connection
{
	public class ConnectionList : CollectionBase
	{
        #region Public Properties
        public ConnectionInfo this[object Index]
		{
			get
			{
				if (Index is ConnectionInfo)
				{
					return (ConnectionInfo)Index;
				}
				else
				{
					return ((ConnectionInfo) (List[Convert.ToInt32(Index)]));
				}
			}
		}
			
        public new int Count
		{
			get { return List.Count; }
		}
        #endregion
			
        #region Public Methods
		public ConnectionInfo Add(ConnectionInfo cInfo)
		{
			this.List.Add(cInfo);
			return cInfo;
		}
			
		public void AddRange(ConnectionInfo[] cInfo)
		{
			foreach (ConnectionInfo cI in cInfo)
			{
				List.Add(cI);
			}
		}
			
		public ConnectionInfo FindByConstantID(string id)
		{
			foreach (ConnectionInfo conI in List)
			{
				if (conI.ConstantID == id)
				{
					return conI;
				}
			}
				
			return null;
		}
			
		public Connection.ConnectionList Copy()
		{
			try
			{
                return (Connection.ConnectionList)this.MemberwiseClone();
			}
			catch (Exception)
			{
			}
				
			return null;
		}
			
		public new void Clear()
		{
			this.List.Clear();
		}
        #endregion
	}
}